frontend/src/pages/AllListings.tsx

```diff
@@ -1,31 +1,82 @@
 import * as React from "react";
+import { Typography, Accordion, AccordionSummary, AccordionDetails } from '@mui/material';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
 import ListingGrid from "src/components/Listing/ListingGrid";
 import PageHeader from "src/components/PageHeader/PageHeader";
 import { LISTING } from "src/constants/endpoints";
 import { useHttpClient } from "src/hooks/http-hook";
 import { buildListings } from "src/utils/listings";
+import { RECOMMENDATION } from "src/constants/endpoints";
+import { UserLoginContext } from "src/context/userLoginContext";
+import RecommendedCarousel from 'src/components/Recommended/RecommendedCarousel'
 
 export interface IAllListingsProps { }
 
 export default function AllListings(props: IAllListingsProps) {
-  const [listings, setListings] = React.useState([]);
-  const { sendRequest } = useHttpClient();
+    const [listings, setListings] = React.useState([]);
+    const [recommendedListings, setRecommendedListings] = React.useState([])
+    const { sendRequest } = useHttpClient();
+    const { isLoggedIn, profile } = React.useContext(UserLoginContext);
+    React.useEffect(() => {
+        const getSellerItems = async () => {
+            let items = await sendRequest(LISTING, "GET", {});
+            items = buildListings(items);
+            setListings(items);
+        };
+        
+        const getRecommendedItems = async () => {
+            if (isLoggedIn) {
+                let recommendations = await sendRequest(RECOMMENDATION,"GET", {}, `/${profile.id}`);
+                recommendations = buildListings(recommendations);
+                setRecommendedListings(recommendations)
+            }
+        };
```

Comment:
Could you move this API call into the RecommendedCarousel component so that the carousel would be more easily reusable without the need for it's parent to API call it? This is probably the only carousel that we will use

Reply:
Thats a good comment. Good job